src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/OneTimeLinkController.kt

```diff
@@ -28,7 +32,43 @@ class OneTimeLinkController(
         val token = context.params(":token")
         val claims = verifyToken(token, repo)
         val link = OneTimeLink.parseClaims(claims)
-        return link.perform(controllers, context, repos)
+        val redirectUrl = link.queryParams["redirectUrl"]
+
+        try
+        {
+            val data = link.perform(controllers, context, repos)
+            return if (redirectUrl.isNullOrEmpty())
+            {
+                data
+            }
+            else
+            {
+                val result = Result(ResultStatus.SUCCESS, data, emptyList())
+                redirectWithResult(context, result, redirectUrl!!)
+            }
+        }
+        catch (e: Exception)
+        {
+            if (redirectUrl.isNullOrEmpty())
+            {
+                throw e
+            }
+
+            val error = when (e)
+            {
+                is MontaguError -> e.asResult()
+                else -> UnexpectedError().asResult()
+            }
```

Comment:
Do errors get logged if we do this? 

In general, is there a way of reusing more of the code in `ErrorHandler` to make sure the behaviour is more similar?